File: java/sort/array/GnomeSort.java
package sort.array;

public class GnomeSort {
    public void sort(int[] nums) {
        int index = 0;
        while (index < nums.length) {
            if (0 == index || nums[index - 1] <= nums[index]) {
                ++index;
            } else {
                nums[index] ^= nums[index - 1];
                nums[index - 1] ^= nums[index];
                nums[index] ^= nums[index - 1];

                --index;
            }
        }
    }
}

File: py/sort/array/GnomeSort.py
#!/usr/bin/env python
# -*- coding:gbk -*-

from typing import List

class GnomeSort:
    def sort(self, nums: List[int]) -> None:
        index = 0
        while index < len(nums):
            if not index or nums[index - 1] <= nums[index]:
                index += 1
            else:
                nums[index], nums[index - 1] = nums[index - 1], nums[index]
                index -= 1
